### PR TITLE
Enhance Best Justified Checkpoint Update Logic

### LINKED ISSUE #639: ensure best_justified_checkpoint updates to the best available
v0.9.3:  (ethereum/eth2.0-specs#1508)

### PR DESCRIPTION
**Pull request by aunyks**
Resolves #639

### CODE REVIEW

File: packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts
Comment (mpetrunic) [inline packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts]: This would probably benefit from clock I'm adding in https://github.com/ChainSafe/lodestar/pull/648/files#diff-c823e58d682201704d876febbedd0d52R4

Comment (aunyks) [reply packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts]: In that case, should I wait for that to be merged before merging this?

Comment (twoeths) [reply packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts]: yeah let's use the new clock in @mpetrunic PR, so please hold on a little bit.

File: packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts
Comment (twoeths) [inline packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts:245]: the logic is right but probably we want to do everything inside `checkAndSetJustified`

Comment (aunyks) [reply packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts:245]: You're talking about the `if` statements surrounding that call to `checkAndSetJustified`, correct?

Comment (twoeths) [reply packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts:245]: yes. I mean we keep the `this.checkAndSetJustified(justifiedCheckpoint);` call and change the logic inside `checkAndSetJustified` because I think it's better to centralize the check and assignment of `bestJustifiedCheckpoint` in one place

Comment (twoeths) [reply packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts:245]: so sorry, I misunderstood you. What I mean is we keep both the `if (justifiedCheckpoint && (!this.justified || justifiedCheckpoint.epoch > this.justified.epoch)) {` and `this.checkAndSetJustified(justifiedCheckpoint);` and just do the requested change inside `checkAndSetJustified`

Comment (aunyks) [reply packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts:245]: I'm not sure I understand. `checkAndSetJustified` is already doing the change, no?

Comment (twoeths) [reply packages/lodestar/src/chain/forkChoice/statefulDag/lmdGhost.ts:245]: yeah it does its job, just want to keep minimal change to the current code but probably this looks good enough too. I may need to change the flow a little bit in my other PR anyway.

**Review by twoeths — APPROVED** [review]

**Pull request merged**
